File: day1/day1-js/day1.js
const fs = require("fs");
const { parse } = require("path");

function increment(input) {
  var increment = 0;
  var array = input.split("\n");
  for (var i = 0; i < array.length - 1; i++) {
    if (parseInt(array[i]) < parseInt(array[i + 1])) {
      increment += 1;
    }
  }
  return increment;
}

function triples(input) {
  count = 0;
  var array = input.split("\n");
  valor = parseInt(array[0]) + parseInt(array[1]) + parseInt(array[2]);
  for (var i = 1; i < array.length - 2; i++) {
    var temp =
      parseInt(array[i]) + parseInt(array[i + 1]) + parseInt(array[i + 2]);
    if (temp > valor) {
      count++;
    }
    valor = temp;
  }
  return count;
}

input = fs.readFileSync("input.txt", "utf8");
console.log("The correct result for question one is:", increment(input));
console.log("The correct result for question two is:", triples(input));

File: day1/day1-dart/bin/day1.dart
import 'dart:io';

void main(List<String> arguments) async {
  final input = File('input.txt');

  input.readAsLines().then((value) =>
      (print("Question 1: ${countIncrements(value.map(int.parse).toList())}")));
  input.readAsLines().then((value) =>
      (print("Question 2: ${countTriples(value.map(int.parse).toList())}")));
}

int countIncrements(List<int> list) {
  int count = 0;

  for (int i = 0; i < list.length - 1; i++) {
    if (list[i] < list[i + 1]) {
      count++;
    }
  }
  print(count);
  return count;
}

int countTriples(List<int> list) {
  int count = 0;
  int max = list[0] + list[1] + list[2];

  for (int i = 0; i < list.length - 2; i++) {
    int temp = list[i] + list[i + 1] + list[i + 2];
    if (temp > max) {
      count++;
    }
    max = temp;
  }

  return count;
}
